src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -102,15 +100,19 @@ class InstanceOpFactoryImpl(
       val maybeVolumeMatch = PersistentVolumeMatcher.matchVolumes(offer, runSpec, request.reserved)
 
       maybeVolumeMatch.flatMap { volumeMatch =>
+
         // we must not consider the volumeMatch's Reserved task because that would lead to a violation of constraints
         // by the Reserved task that we actually want to launch
-        val tasksToConsiderForConstraints = tasks.filter(_.taskId != volumeMatch.task.taskId)
+        val instancesToConsiderForConstraints = instances.values.filter { inst =>
+          inst.tasks.filter(_.taskId != volumeMatch.task.taskId).nonEmpty
+        }.toVector
```

Comment:
I have a new cleanup change that's going to rip out JavaCoverters and JavaConversions in favor of minimally copied collections (eg, go directly to the target type). Furthermore, if we transform the collection, the transform will be performed in the java8 stream API and "collected" into the Scala collection instead. I'm still working through how to make this not ugly, eg, I may make a rich stream that understands Scala types and lambas better (since we don't have 2.12 yet)